### PR TITLE
[a11y] Tab action order for `FAQ Groups` according to specification

### PR DESCRIPTION
**Pull request by stamat**
## Summary

This PR contains a fix update to `FAQ Groups` tab order accessibility issue 👇 . Tab list items should be navigateable by arrow keys and only one element should be focusable, based on this specification https://www.w3.org/WAI/ARIA/apg/patterns/tabs/ 

- Closes: https://github.com/github/accessibility-audits/issues/7964

I would love to ship this ASAP 🙏 

## List of notable changes:

- **added** arrow up and down navigation of vertical tabs
- **added** tabs are no `tabindex="-1"` except the active one - this fixes the a11y tab action order issue

## What should reviewers focus on?

- Review added functionality, it shoul completely align with the https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

## Steps to test:

1. Navigate to `/story/components-faq-features--groups` story in the Storybook
2. Pressing the <kbd>Tab</kbd> key and <kbd>Arrow Up</kbd> and <kbd> Arrow Down </kbd>, confirm that the navigation of the tabs is according to specification https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

## Supporting resources (related issues, external links, etc):

- https://github.com/github/accessibility-audits/issues/7964#issuecomment-2123542525
- https://www.w3.org/WAI/ARIA/apg/patterns/tabs/

## Contributor checklist:

- [x] All new and existing CI checks pass
- [x] Tests prove that the feature works and covers both happy and unhappy paths
- [x] Any drop in coverage, breaking changes or regressions have been documented above
- [x] New visual snapshots have been generated / updated for any UI changes
- [x] All developer debugging and non-functional logging has been removed
- [x] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

**Review by fisanchez — APPROVED** [review]
It's always great to see A11y changes!

**Review by joshfarrant — APPROVED** [review]
Nice, works perfectly!

File: .changeset/beige-kings-allow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@primer/react-brand": patch
+---
+
+[a11y] Tab action order for `FAQ Groups` according to specification
```
Comment (rezrah) [inline .changeset/beige-kings-allow.md:5]: Adding some verbose details for the changelog.

```suggestion
Improved keyboard navigation for tab lists in the `FAQ Groups` component. Now features focus trapping, and `up` and `down` keypress navigation as specified in the ARIA tab panel pattern.
```

File: packages/react/src/FAQ/FAQGroup.tsx [hunk 2]
```diff
@@ -60,8 +83,11 @@ function _FAQGroup({children, id, defaultSelectedIndex = 0, ...rest}: FAQGroupPr
           aria-controls={`${instanceId}-panel-${index}`}
           aria-selected={selectedIndex === index}
           onClick={handleTabClick(index)}
+          onKeyDown={handleKeyDown(index)}
           key={index}
           data-testid={`FAQGroup-tab-${index + 1}`}
+          tabIndex={selectedIndex === index ? null : -1}
+          ref={selectedIndex === index ? selectedTabRef : null}
```
Comment (rezrah) [inline packages/react/src/FAQ/FAQGroup.tsx:90]: Just inverting the condition here to put truthy value first...

```suggestion
          tabIndex={selectedIndex !== index ? -1 : undefined}
          ref={selectedIndex === index ? selectedTabRef : undefined}
```

Comment (rezrah) [reply packages/react/src/FAQ/FAQGroup.tsx]: Fixed in [5c9b95c](https://github.com/primer/brand/pull/596/commits/5c9b95c462211e48f80224534a07ca0cc5de04c7)

File: .changeset/beige-kings-allow.md (hunk 1 above)
Comment (rezrah) [reply .changeset/beige-kings-allow.md]: Fixed in [5c9b95c](https://github.com/primer/brand/pull/596/commits/5c9b95c462211e48f80224534a07ca0cc5de04c7)

**Review by rezrah — APPROVED** [review]
Thanks for fix @stamat.

FYI, this is now ready to merge...

I've made the following changes to this PR in [5c9b95c](https://github.com/primer/brand/pull/596/commits/5c9b95c462211e48f80224534a07ca0cc5de04c7)

- Added unit tests
- Regen'd snapshots and removed randomisation of images on an unrelated story, which was causing the VRT warnings in this PR
- Updated some conditional logic for project consistency

**Pull request merged**